tools/log-collector/helper.go

```diff
@@ -296,3 +299,97 @@ func MatchExpressions(objLabels map[string]string, matchExpr []apiv1.LabelSelect
 	}
 	return true
 }
+
+// getPVCsUsedByPods returns a set of PVC namespaced names used by the given pods
+func getPVCsUsedByPods(pods []unstructured.Unstructured) map[types.NamespacedName]bool {
```

Comment:
handle errors properly across all functions in this file. In case of strict parsing where spec or names are required from the object but not found, log a warning and then continue.